server/src/main/java/org/elasticsearch/cluster/routing/allocation/DiskThresholdSettings.java

```diff
@@ -47,6 +61,20 @@ public class DiskThresholdSettings {
         Setting.Property.Dynamic,
         Setting.Property.NodeScope
     );
+    public static final Setting<ByteSizeValue> CLUSTER_ROUTING_ALLOCATION_HIGH_DISK_MAX_HEADROOM_SETTING = new Setting<>(
+        "cluster.routing.allocation.disk.watermark.high.max_headroom",
+        (settings) -> {
+            if (CLUSTER_ROUTING_ALLOCATION_HIGH_DISK_WATERMARK_SETTING.exists(settings)) {
+                return "-1";
+            } else {
+                return "100GB";
```

Comment:
Hi @DaveCTurner ! I believe you may think that the max headroom settings replaces the percentage high watermark setting. No. It only possibly caps the minimum free amount, rather than increasing it. Seeing the DiskThresholdSettingsTests::testDefaults() test should make it easier to see:
* For small disks, like 100 bytes, the default percentages apply. So e.g. the high watermark demands 10 bytes of free space (rather than 100GB of the default max headroom).
* For very large disks, like 1000 TB, the default high percentage would require 100 TB of free space, but the default high max headroom setting caps this to 100GB. So instead of having unassigned shards at below 100TB free space, you have unassigned shards at below 100GB of free space.
I hope this clarifies it. If not, please unresolve this conversation.

Reply:
Ah you're right, sorry, I was thinking about the problems at the other end of the scale for other reasons.